app/src/main/java/currencyConverter/repository/ICountryRepository.java

```diff
@@ -0,0 +1,15 @@
+package currencyConverter.repository;
+
+import currencyConverter.exception.CountryFethchingException;
+import currencyConverter.model.CountryModel;
+
+public interface ICountryRepository {

```

Comment:
I haven't seen any use of this repository in code. Let`s create a service that will combine repository and provider classes and make them both work together. In that case we will be able to make those classes effectively private and to provide a new interface to a client code (just because client code should't worry about model creation/saving/updating). Moreover, in this case, we will have a small automatizated library, that would manage everything connected with country by itself.